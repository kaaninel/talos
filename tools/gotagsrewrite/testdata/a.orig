package testdata

//gotagsrewrite:gen
type MySpecSpec struct {
	NewField     string `yaml:"new_field"`
	FirstField   string
	Name         string `yaml:"string"`
	unexported   string
	AnotherField string `yaml:"another_field" protobuf:"3"`
	LastField    string
}

type MySpec struct{}

type MySpecSpec2 struct {
	unexported string
	MySpec
	unexported2 string
}

type MySpecSpec3 struct {
	NewField string `yaml:"new_field"`
}

// MyCustomStruct is a custom struct
//gotagsrewrite:gen
type MyCustomStruct struct {
	NewField string `yaml:"new_field"`
}

type (
	//gotagsrewrite:gen
	MyAnotherStruct struct {
		NewField string `yaml:"new_field"`
	}

	//gotagsrewrite:gen
	MyBasicStruct struct {
		NewField string `yaml:"new_field"`
	}

	MyNoneStruct struct {
		NewField string `yaml:"new_field"`
	}
)

//gotagsrewrite:gen
type (
	MyOnlyStruct struct {
		NewField string `yaml:"new_field"`
	}
)

//gotagsrewrite:gen
type (
	MyOnlyStruct2 struct {
		NewField string `yaml:"new_field"`
	}

	MyOnlyStruct3 struct {
		NewField string `yaml:"new_field"`
	}
)
