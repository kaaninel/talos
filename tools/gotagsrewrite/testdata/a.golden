package testdata

//gotagsrewrite:gen
type MySpecSpec struct {
	NewField     string `yaml:"new_field" protobuf:"4"`
	FirstField   string `protobuf:"5"`
	Name         string `yaml:"string" protobuf:"6"`
	unexported   string
	AnotherField string `yaml:"another_field" protobuf:"3"`
	LastField    string `protobuf:"7"`
}

type MySpec struct{}

type MySpecSpec2 struct {
	unexported string
	MySpec
	unexported2 string
}

type MySpecSpec3 struct {
	NewField string `yaml:"new_field"`
}

// MyCustomStruct is a custom struct
//
//gotagsrewrite:gen
type MyCustomStruct struct {
	NewField string `yaml:"new_field" protobuf:"1"`
}

type (
	//gotagsrewrite:gen
	MyAnotherStruct struct {
		NewField string `yaml:"new_field" protobuf:"1"`
	}

	//gotagsrewrite:gen
	MyBasicStruct struct {
		NewField string `yaml:"new_field" protobuf:"1"`
	}

	MyNoneStruct struct {
		NewField string `yaml:"new_field"`
	}
)

//gotagsrewrite:gen
type (
	MyOnlyStruct struct {
		NewField string `yaml:"new_field" protobuf:"1"`
	}
)

//gotagsrewrite:gen
type (
	MyOnlyStruct2 struct {
		NewField string `yaml:"new_field"`
	}

	MyOnlyStruct3 struct {
		NewField string `yaml:"new_field"`
	}
)
